experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -290,3 +290,130 @@ function parabolic_subgroup_with_projection(
   end
   return factor, emb, MapFromFunc(W, factor, proj)
 end
+
+@doc raw"""
+    irreducible_factors(W::WeylGroup; morphisms::Bool=false)
+
+If `morphisms` is `true`, return a triple (`U`, `emb`, `proj`) that describes the irreducible factors of `W`.
+That is, `U`, `emb`, `proj` are vectors whose length is
+the number of irreducible components of the Dynkin diagram of `W`, and for each `i`,
+(`U[i]`, `emb[i]`, `proj[i]`) describes the `i`-th factor of `W` as described in
+[`parabolic_subgroup_with_projection`](@ref).
+
+If `morphisms` is `false`, return only `U` .
+
+The order of the irreducible factors is the one given by [`cartan_type_with_ordering`](@ref).
+
+See also [`inner_direct_product(::AbstractVector{WeylGroup})`](@ref).
+
+# Examples
+```jldoctest
+julia> W = weyl_group([(:A, 3), (:B, 4)]);
+
+julia> irreducible_factors(W)
+2-element Vector{WeylGroup}:
+ Weyl group of root system of type A3
+ Weyl group of root system of type B4
+
+julia> U, emb, proj = irreducible_factors(W; morphisms=true)
+(WeylGroup[Weyl group of root system of type A3, Weyl group of root system of type B4], Map{WeylGroup, WeylGroup}[Map: Weyl group -> W, Map: Weyl group -> W], Map{WeylGroup, WeylGroup}[Map: W -> Weyl group, Map: W -> Weyl group])
+
+julia> emb[1](U[1][1]) == W[1]
+true
+
+julia> proj[2](W[4]) == U[2][1]
+true
+```
+"""
+function irreducible_factors(W::WeylGroup; morphisms::Bool=false)
+  types, ordering = root_system_type_with_ordering(root_system(W))
+  num_factors = length(types) # Number of irreducible factors of W
+
+  U = Vector{WeylGroup}(undef, num_factors)
+  emb = Vector{Map{WeylGroup,WeylGroup}}(undef, num_factors)
+  proj = Vector{Map{WeylGroup,WeylGroup}}(undef, num_factors)
+
+  start_index = 1
+  for (i, type) in enumerate(types)
+    rk = type[2] # rank of the i-th irreducible factor
+    end_index = start_index + rk - 1
+    U[i], emb[i], proj[i] = parabolic_subgroup_with_projection(
+      W, ordering[start_index:end_index]; check=false # No check needed by construction
+    )
+    start_index = end_index + 1
+  end
+
+  if morphisms
+    return U, emb, proj
+  else
+    return U
+  end
+end
+
+@doc raw"""
+    inner_direct_product(L::AbstractVector{WeylGroup}; morphisms::Bool=false)
+    inner_direct_product(L::WeylGroup...; morphisms::Bool=false)
+
+If `morphisms` is `false`, then return a Weyl group `W` that is isomorphic to the
+direct product of the Weyl groups in `L`.
+The generators of `W` are in natural 1-1 correspondence with the generators of the groups in `L`
+in the given order.
+
+If `morphisms` is `true`, return a triple (`W`, `emb`, `proj`) where
+`W` is as above and `emb`, `proj` are vectors such that
+`emb[i]` (resp., `proj[i]`) is the embedding of `L[i]` into `W`
+(resp., the projection of `W` onto `L[i]`).
+
+See also [`inner_direct_product(::AbstractVector{T})`](@ref).
```

Comment:
```suggestion
See also [`inner_direct_product(::AbstractVector{T}) where {T<:Union{PcGroup, SubPcGroup, FPGroup, SubFPGroup}}`](@ref).
```
Otherwise, this won't reference the correct docstring

Reply:
With this change, I get the following error when building the documentation:
```
ERROR: TypeError: in typeassert, expected DataType, got Type{Union{Tuple{AbstractArray{T<:Union{FPGroup, PcGroup, SubFPGroup, SubPcGroup}, 1}}, Tuple{T<:Union{FPGroup, PcGroup, SubFPGroup, SubPcGroup}}}}
Stacktrace:
  [1] _hasmethod(f::Any, t::Any)
    @ Base ./Base.jl:192
  [2] hasmethod(f::Any, t::Any)
    @ Base ./reflection.jl:2347
  [3] find_object(λ::typeof(inner_direct_product), binding::Base.Docs.Binding, typesig::Type)
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:510
  [4] find_object(binding::Base.Docs.Binding, typesig::Type)
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:504
  [5] find_object(doc::Documenter.Document, binding::Base.Docs.Binding, typesig::Type)
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:490
  [6] docsxref(node::MarkdownAST.Node{Nothing}, code::String, meta::Dict{Symbol, Any}, page::Documenter.Page, doc::Documenter.Document, errors::Vector{String})
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:402
  [7] runner(::Type{Documenter.XRefResolvers.Docs}, node::MarkdownAST.Node{Nothing}, slug::String, meta::Dict{Symbol, Any}, page::Documenter.Page, doc::Documenter.Document, errors::Vector{String})
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:255
  [8] dispatch(::Type{Documenter.XRefResolvers.XRefResolverPipeline}, ::MarkdownAST.Node{Nothing}, ::Vararg{Any})
    @ Documenter.Selectors ~/.julia/packages/Documenter/iwb7N/src/utilities/Selectors.jl:170
  [9] xref(node::MarkdownAST.Node{Nothing}, meta::Dict{Symbol, Any}, page::Documenter.Page, doc::Documenter.Document)
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:305
 [10] crossref(doc::Documenter.Document, page::Documenter.Page, mdast::MarkdownAST.Node{Nothing})
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:49
 [11] crossref(doc::Documenter.Document, page::Documenter.Page, mdast::MarkdownAST.Node{Nothing})
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:36
 [12] crossref(doc::Documenter.Document)
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/cross_references.jl:17
 [13] runner(::Type{Documenter.Builder.CrossReferences}, doc::Documenter.Document)
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/builder_pipeline.jl:231
 [14] dispatch(::Type{Documenter.Builder.DocumentPipeline}, x::Documenter.Document)
    @ Documenter.Selectors ~/.julia/packages/Documenter/iwb7N/src/utilities/Selectors.jl:170
 [15] #89
    @ ~/.julia/packages/Documenter/iwb7N/src/makedocs.jl:273 [inlined]
 [16] withenv(::Documenter.var"#89#91"{Documenter.Document}, ::Pair{String, Nothing}, ::Vararg{Pair{String, Nothing}})
    @ Base ./env.jl:265
 [17] #88
    @ ~/.julia/packages/Documenter/iwb7N/src/makedocs.jl:272 [inlined]
 [18] cd(f::Documenter.var"#88#90"{Documenter.Document}, dir::String)
    @ Base.Filesystem ./file.jl:112
 [19] makedocs(; debug::Bool, format::Documenter.HTMLWriter.HTML, kwargs::@Kwargs{…})
    @ Documenter ~/.julia/packages/Documenter/iwb7N/src/makedocs.jl:271
 [20] (::Main.BuildDoc.var"#3#5"{Bool, Bool, Bool, Module, String, String, String, String, DocumenterCitations.CitationBibliography, Vector{Any}})()
    @ Main.BuildDoc ~/Oscar.jl/docs/make_work.jl:183
 [21] cd(f::Main.BuildDoc.var"#3#5"{Bool, Bool, Bool, Module, String, String, String, String, DocumenterCitations.CitationBibliography, Vector{Any}}, dir::String)
    @ Base.Filesystem ./file.jl:112
 [22] doit(Oscar::Module; warnonly::Bool, local_build::Bool, doctest::Bool)
    @ Main.BuildDoc ~/Oscar.jl/docs/make_work.jl:173
 [23] #invokelatest#2
    @ ./essentials.jl:1057 [inlined]
 [24] invokelatest
    @ ./essentials.jl:1052 [inlined]
 [25] (::Oscar.var"#29#32"{Bool, Bool})()
    @ Oscar ~/Oscar.jl/src/utils/docs.jl:228
 [26] activate(f::Oscar.var"#29#32"{Bool, Bool}, new_project::String)
    @ Pkg.API ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:1379
 [27] (::Oscar.var"#28#31"{Bool, Bool})()
    @ Oscar ~/Oscar.jl/src/utils/docs.jl:227
 [28] with_unicode(f::Oscar.var"#28#31"{Bool, Bool}, allowed::Bool)
    @ AbstractAlgebra.PrettyPrinting ~/.julia/packages/AbstractAlgebra/lYBCD/src/PrettyPrinting.jl:1846
 [29] #27
    @ ~/Oscar.jl/src/utils/docs.jl:226 [inlined]
 [30] withenv(::Oscar.var"#27#30"{Bool, Bool}, ::Pair{String, Int64}, ::Vararg{Pair{String, Int64}})
    @ Base ./env.jl:265
 [31] build_doc(; doctest::Bool, warnonly::Bool, open_browser::Bool, start_server::Bool)
    @ Oscar ~/Oscar.jl/src/utils/docs.jl:225
 [32] top-level scope
    @ REPL[14]:1
Some type information was truncated. Use `show(err)` to see complete types.
```
If I remove this line (`WeylGroup.jl::367`), the documentation builds successfully. I thought it might be because the docstring of the other `inner_direct_product` only contains
```
    inner_direct_product(L::AbstractVector{T}; morphisms)
```
without the `where` statement, but I could not get it to work by changing the old docstring either. I also do not really understand what the error message is saying. Could you please have a look?

Reply:
Hmm, that didn't help. I am trying some things locally now

Reply:
I don't know why my change just now fixed it (at least locally), but I does work :)
Seems to be related to https://github.com/JuliaLang/julia/issues/29437 somehow (cc @thofma, ref https://github.com/thofma/WhereIsMyDocstring.jl/issues/11)

Reply:
Thanks! Works (locally) for me as well.